rnote-cli/src/cli.rs

```diff
@@ -70,18 +74,51 @@ pub(crate) async fn run() -> anyhow::Result<()> {
             import_file(&mut engine, input_file, rnote_file).await?;
         }
         Commands::Export {
-            rnote_file,
+            rnote_files,
             output_file,
+            output_format,
             with_background,
             with_pattern,
         } => {
             println!("Exporting..");
 
-            // apply given arguments to export prefs
-            engine.export_prefs.doc_export_prefs =
-                create_doc_export_prefs_from_args(&output_file, with_background, with_pattern)?;
+            let format = match (&output_file, output_format) {
+                (Some(file), None) => {
+                    get_export_format(file.extension().and_then(|ext| ext.to_str()))
+                }
+                (_, out_format) => get_export_format(out_format.as_deref()),
+            };
+
+            match format {
+                Ok(format) => {
+                    // apply given arguments to export prefs
+                    engine.export_prefs.doc_export_prefs =
+                        create_doc_export_prefs_from_args(format, with_background, with_pattern)?;
+                }
+                Err(err) => return Err(err),
+            }
```

Comment:
I think determining the format should be moved into `create_doc_export_prefs_from_args`.

Reply:
well you could use it from the engine after it has been applied to it

Reply:
ooh ok got it, didn't see the `file_ext()` function in `DocExportFormat`